apps/aecore/lib/aecore/account/account.ex

```diff
@@ -343,6 +343,20 @@ defmodule Aecore.Account.Account do
     %Account{account_state | balance: new_balance}
   end
 
+  @doc """
+  Adds balance to the given Account state. No exception version.
+  """
+  @spec apply_transfer(Account.t(), non_neg_integer(), integer()) :: {:ok, Account.t()} | error()
```

Comment:
why is this function added? if this is a variant of another one, we should use one variant for all use-cases

Reply:
@thepiwo The `apply_transfer!` variant uses exceptions to handle errors and is used in the on chain transaction processing stack. The off chain updates need to change the state of accounts and originally used the version with exceptions(https://github.com/aeternity/elixir-node/pull/586/commits/d8fb2596a57872dabbeae88348bd1750e48fb689#diff-850c2fc5dd4f310c3b6b7fa770e16236L128). Handling exceptions is slow so it was suggested by @cytadela8 to create an exception-less version. 

We can consider to go back to the version which uses exceptions or refactor the onchain transactions so they switch to the exception-less version(this can be a separate issue).

Reply:
I prefer the exception-less functions over the existing ones, could we use them for all cases? Otherwise we could keep it like it is, but refactor so the exception-less version gets called by the `!` one.